include/aspect/simulator.h

```diff
@@ -1337,6 +1337,14 @@ namespace aspect
       void interpolate_onto_velocity_system(const TensorFunction<1,dim> &func,
                                             LinearAlgebra::Vector &vec) const;
 
+      /**
+       * Perform a newton line search to determine the optimal step length
+       * along the search direction.
```

Comment:
Also say that the result of the function is that `current_linearization_point` is set to the old `current_linearization_point` plus a suitable update in direction of `search_direction`.

Reply:
You'll still want to address this point.